radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/steady_state/simulation/full_function_forks/CoordinatedForkSanityTest.java

```diff
@@ -216,8 +217,8 @@ public void sanity_tests_should_pass() {
 
   private void updateValidatorWithLatestFork(RunningNetwork network, BFTNode node) {
     final var forks = network.getInstance(Forks.class, node);
-    final var maybeForkVoteHash =
-        forks.getCandidateFork().map(f -> ForkConfig.voteHash(node.getKey(), f));
+    final var maybeForkVote =
```

Comment:
I would remove "maybe"from this variable as there's no Maybe type in Java afaik.

Reply:
I prefer to make it clear when the var is optional, I suggest either `maybeXyz` or `xyzOpt`

Reply:
I'm not sure this is a common pattern in Java, when using an IDE we can easily hover the name or use auto complete and see this is an `Optional` type. This is something we can discuss as a team.  But in the case of using it, I believe it is better to use a name similar to the underlying type.

Reply:
yup, let's discuss

Reply:
The goal here isn't really to provide the information about the type, but rather to capture what the value really is.
If the names was just `forkVote` it would mean that the value represents a fork vote, which is not always true. The value can represent either a fork vote or lack of thereof. The type doesn't matter, it could be any container (Optional, Maybe, etc) or an ADT (like an enum [if Java had something like that] ` { ForkVote(HashCode value), NoVote }`) or it could be a plain nullable value - in all these cases I'd still like to capture this information in the name of the variable (with either a `maybe` prefix or `Opt` suffix).

Reply:
also, it removes the name conflict if at some later point you want to extract it to a non-nullable value:
```
val xyzOpt = ...
val xyz = xyzOpt.orElse(defaultXyz);
```

Reply:
That's exactly what an `Optional<CandidateForkVote>` conveys, either a fork vote or nothing.

Reply:
following this reasoning, can I just call it `potato` since the type already conveys the meaning, why duplicate the same thing?

Reply:
I think it seems weird to me cause I've never seen anyone using "possibleValue" to name a variable which represents the result of getting a value from a map in Java, which can return null, for example. Seems this need only arises when using an `Optional` type which is not that different conceptually, only more explicit and safer.

Reply:
I think not naming things correctly is one of the reasons why null reference is considered a billion-dollar mistake ;) Imagine how much less likely it would be to miss a null check if the name clearly stated that the value might or might not be there. I'd definitely go for `maybeXyz` (or `xyzOpt`) even for nullable values, but luckily we have Optional.